conbench/app/_plots.py

```diff
@@ -413,59 +440,79 @@ def gen_js_callback_click_on_glyph_show_run_details(repo_string):
+                    [s.commit_timestamp.isoformat() for s in samples]
+                ),
                 dist_change_indexes,
             ),
             # TODO: best-case is not always min, e.g. when data has a unit like
-            # bandwidth.
-            y=_insert_nans([min(x["data"]) for x in history], dist_change_indexes),
+            # bandwidth. Note(JP): min() must not have None passed as arg, that
+            # is why I have changed `s.data` to only contain type float
+            # elements (including math.nan, instead of None).
+            y=_insert_nans([min(s.data) for s in samples], dist_change_indexes),
         )
     )
 
+    dummy_hs = HistorySample(
+        mean=benchmark["stats"]["mean"],
+        commit_msg=run["commit"]["message"],
+        commit_timestamp=util.tznaive_iso8601_to_tzaware_dt(run["commit"]["timestamp"]),
+        commit_hash=run["commit"]["sha"],
+        benchmark_result_id=benchmark["id"],
+        repository=run["commit"]["repository"],
+        # Right now, the _source() function requires as first arg a list of
+        # HistorySample objects. Comply with this, but most of the info is
+        # ignored. We may want to add a new type for stream-lining this.
+        case_id="dummy",  # not consumed
+        context_id="dummy",  # not consumed
+        data=[0],  # not consumed
+        times=[0],  # not consumed
+        unit="dummy",  # not consumed
+        hardware_hash="dummy",  # not consumed
+        run_name="dummy",  # not consumed
+        zscorestats=HistorySampleZscoreStats(
+            begins_distribution_change=False,  # not consumed
+            segment_id="dummy",  # not consumed
+            rolling_mean_excluding_this_commit=0.0,  # not consumed
+            rolling_mean=0.0,  # not consumed
+            residual=0.0,  # not consumed
+            rolling_stddev=0.0,  # not consumed
+        ),
+    )
+
     source_current_bm_mean = _source(
-        [
-            {
-                "mean": benchmark["stats"]["mean"],
-                "message": run["commit"]["message"],
-                "timestamp": run["commit"]["timestamp"],
-                "sha": run["commit"]["sha"],
-                "benchmark_id": benchmark["id"],
-                "repository": run["commit"]["repository"],
-            }
-        ],
+        [dummy_hs],
         unit,
         formatted=formatted,
     )
 
+    # create same dummy structure, only difference: set min as mean.
+    dummy_hs_min = copy.copy(dummy_hs)
+    dummy_hs_min.mean = benchmark["stats"]["min"]
     source_current_bm_min = _source(
-        [
-            {
-                "mean": benchmark["stats"]["min"],
-                "message": run["commit"]["message"],
-                "timestamp": run["commit"]["timestamp"],
-                "sha": run["commit"]["sha"],
-                "benchmark_id": benchmark["id"],
-                "repository": run["commit"]["repository"],
-            }
-        ],
+        [dummy_hs_min],
```

Comment:
Now this is the part where I took a quick and dirty way out of the situation. `source_current_bm_mean` and `source_current_bm_min` are one-sample curves and the actually relevant input is `benchmark["stats"]["mean"]` and `benchmark["stats"]["min"]`, respectively. For these one-sample situations I believe we may want to construct the ColumnarDataSource objects directly, and not go through _source().